src/leap/keymanager/__init__.py

```diff
@@ -219,13 +220,21 @@ def _fetch_keys_from_server(self, address):
             res = self._get(self._nickserver_uri, {'address': address})
             res.raise_for_status()
             server_keys = res.json()
+
             # insert keys in local database
             if self.OPENPGP_KEY in server_keys:
+                # nicknym server is authoritative for its own domain,
+                # for other domains the key might come from key servers.
+                validation_level = ValidationLevel.Weak_Chain
+                if (address.split("@")[1] ==
+                        urlparse(self._nickserver_uri).hostname):
```

Comment:
Agree, I was searching for a function in python standard library for that, but I didn't find it. I'll write a helper.